### ISSUE TITLE
New layout of Hanshow Nebular 350

### ISSUE THREAD

**Issue by teichtaucher** [atc1441/ATC_TLSR_Paper#50]
Recently I got an 3.5" hanshow nebular pricetag. Orignally I wanted to reflash it with alternative firmware. But after some cuts into the back of case I found out that they changed the layout - the soldering pads for reflashing are not there as shown in this video:
https://www.youtube.com/watch?v=WRfZJ4xyYwc
Instead it looks like this:
![IMG_20240425_130927](https://github.com/atc1441/ATC_TLSR_Paper/assets/63404380/21af3b44-4e06-4dff-b7ac-b556640cf4b2)
Besides this other layout the also exchanged the battery type. Now it seems that theres a kind of LiPo instead of the two button cell batteries.
Nevertheless, did anybody of you already manage to reflash this new layout?

**Comment by MarquezBU** [comment]
Why destroy the casing so badly?
Simply loosen the plastic cover carefully at the front and everything can be easily removed.

And, your tag is probably about a 350R-N while the video is about the 350Y-N.
But there are no specific differences in functionality or technical specifications between the 350R-N and 350Y-N, except that the 350R-N can display Red, White and Black (BWR) colors.

https://retail-optimiser.de/en/aldi-sued-introduces-esls-from-hanshow/

Check:
https://github.com/atc1441/ATC_TLSR_Paper

atc1441 added support for 3 Color ESL BWR

**Comment by teichtaucher** [comment]
You're right, it's a 350R-N. Originally I wanted to open the casing to connect to the pads to reflash the custom firmware (like this https://youtu.be/WRfZJ4xyYwc?t=153). But as I didn't found them at the expected position I opened the casing even more. And then I found out that the layout looks different and I didn't know how to connect the UART flasher to flash the custom firmware. Any idea?

**Comment by MarquezBU** [comment]
![350R-N_01](https://github.com/atc1441/ATC_TLSR_Paper/assets/129670786/0e7af8ff-ca14-44fb-bac5-a1118ba032d2)
![350R-N_03](https://github.com/atc1441/ATC_TLSR_Paper/assets/129670786/3dfd988b-b55a-46d6-ac71-18702e8cfc5c)
![350R-N_02](https://github.com/atc1441/ATC_TLSR_Paper/assets/129670786/59d255bd-0a7a-4bec-b47e-2e658850ca52)

**Comment by MarquezBU** [comment]
Nicht vergessen die Schraube zu lösen... ; - )

**Comment by teichtaucher** [comment]
Danke, vielleicht zerlege ich meins auch noch mal ordentlicher. Aber mein Kernproblem löst das allerdings nicht. So wie auf dem zweiten Bild sollte es bei mir auch aussehen - oben links ist die LED, und die Anschlüsse für die Batterie sind auch identisch. Was bei mir aber fehlt sind die Anschlüsse zum Flashen über UART. Die sollten bei mir auf dem Bild am oberen Bildrand liegen.

**Comment by atc1441** [comment]
They switch the PCB designs often, even in the video i show 2 versions,

Here its expected that the flashing connector is more in the middle at the thin part(The PCB is very big in your type), here because you got a pouch cell and not CR2450 Batteries

Viel glück

**Comment by MarquezBU** [comment]
Du brauchst einen USB to UART Converter.
https://github.com/atc1441/ATC_TLSR_Paper/blob/main/USB_UART_Flashing_connection.jpg

Mach dein Teil mal sauber von vorne auf. Dann Schraube finden und Platine vorsichtig lösen.
Pins sind auf der Rückseite und beschriftet NRST (RTS) TX (TXD) RX (RXD) SWS VCC/GND

**Comment by teichtaucher** [comment]
Thanks for the the hints! Finally I opened up the ESL on the front, but in my version it's quite difficult because the pouch cell is glued on the PCB and to the casing. So it's very hard to get the pcb out without breaking the glass of the display. Furthermore there are some pins are underneath the pouch cell:
![IMG_20240429_112000](https://github.com/atc1441/ATC_TLSR_Paper/assets/63404380/c8082348-b0b2-4ec7-b53e-5f112bb60bfc)
But I'm not sure if they can be used for flashing. For me it seems that it's more to check the signals and the connection to the display (I also don't see RxD and TxD).
I also tried to use the connection pads I showed in my last image, but it didn't work to get a connection to the controller.
Maybe I will check if I get another PCB revision of Hanshow Nebular 350 or get a Stellar device.

**Comment by teichtaucher** [comment]
OK, finally found the Pins foe reflashing:
![IMG_20240429_114915](https://github.com/atc1441/ATC_TLSR_Paper/assets/63404380/cffc1cac-9192-4420-aa2b-2bd2ae805d96)
It's below the glue sticker at the backside of the PCB. I will now try to flash and Report my status. Hope this helps others.

**Comment by teichtaucher** [comment]
Status update:
I could flash the Nebular 3.5" ESL with some restrictions:

- I had to connect also VDD PIN with the USB2UART. Otherwise the LED was not flashing
- TLSR_HS_BWY_35_inverse.bin was working. I also tried TLSR_HS_BWY_35_ALPHA.bin but it shows a black screen with white letters (I would expect this from TLSR_HS_BWY_35_inverse.bin)
- The display shows the content as expected. But the diplay broke during getting it out of the case. So it could not show the content in a clear manner.
- I tried WebBluetooth Image Uploader to upload an image but the ESL could not get found from my PC. On the display of the ESL  "No AP found" is written. What do I need to do to get an connection and to upload an image?

For the next device I get I will check the model number and if it's the same I will just do a cut in the back of the casing where the flash pins are located. This reduces the risk of damaging the display.

**Comment by ixy05** [comment]
Hi @teichtaucher 

From my investigation into this repo and my own experimentation with these devices (Nebular BWR350), you are trying to do two related but different things. The firmware (TLSR_HS_BWY_35_inverse.bin) is from a different repo https://github.com/OpenEPaperLink/OpenEPaperLink. That firmware is set to work with APs that you have to make. This repo works when you upload the .bin file (found here https://github.com/atc1441/ATC_TLSR_Paper/blob/main/Firmware/ATC_Paper.bin).

I may be wrong or this information is not suitable for your experimentations.

**Comment by teichtaucher** [comment]
Hi @ixy05 ,
thanks for the hint. Do you know if the bin from @atc1441 's repo (https://github.com/atc1441/ATC_TLSR_Paper/blob/main/Firmware/ATC_Paper.bin) supports control by bluetooth? I stopped further investigation as the step to built up an AP is too much for me. I decided to wait until BLE AP can be used to control Hanshow Nebular 350.
